### PR TITLE
adjust MaybeUninit API to discussions

### PR DESCRIPTION
**Pull request by RalfJung**
uninitialized -> uninit
into_initialized -> assume_init
read_initialized -> read
set -> write

### CODE REVIEW

**Comment by jethrogb** [comment]
Missed some in

* src/libstd/sys/sgx/rwlock.rs
* src/libstd/sys/sgx/ext/arch.rs

**Comment by RalfJung** [comment]
Thanks for pointing that out!

The one in rwlock.rs actually already did not compile, it still used `into_inner`...

File: src/libcore/mem.rs [hunk 1]
```diff
@@ -1148,10 +1148,19 @@ impl<T> MaybeUninit<T> {
...
     #[inline(always)]
-    pub const fn uninitialized() -> MaybeUninit<T> {
+    pub const fn uninit() -> MaybeUninit<T> {
         MaybeUninit { uninit: () }
     }
 
+    /// Deprecated before stabilization.
+    #[unstable(feature = "maybe_uninit", issue = "53491")]
+    #[inline(always)]
+    // FIXME: still used by stdsimd
```
Comment (sfackler) [inline src/libcore/mem.rs]: We should be able to future-deprecate it even though stdsimd uses it, right?

Comment (RalfJung) [reply src/libcore/mem.rs]: We have `warn(deprecated_in_future)` and also on CI compiled with `-D warnings`. So I'm afraid I don't think we can.

**Comment by RalfJung** [comment]
Rebased.

@bors r=sfackler

**Comment by jethrogb** [comment]
```
Testing std stage1 (x86_64-unknown-linux-gnu -> x86_64-fortanix-unknown-sgx)
   Compiling std v0.0.0 (/home/jenkins/workspace/rust-sgx-ci/rust/src/libstd)
error: use of deprecated item 'core::mem::MaybeUninit::<T>::set': use `write` instead
   --> src/libstd/sys/sgx/rwlock.rs:271:18
    |
271 |             init.set(RWLock::new());
    |                  ^^^
    |
= note: `-D deprecated` implied by `-D warnings`
```

**Comment by RalfJung** [comment]
Sorry, seems like I missed one use. :/

**Pull request merged**